src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,522 @@
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Presenter;
+
+use Gender;
+use Order;
+use PrestaShop\Module\DemoViewOrderHooks\Entity\Signature;
+
+class SignaturePresenter
+{
+    /**
+     * @var string
+     */
+    private $signatureImgDir;
+
+    public function __construct(string $signatureImgDir)
+    {
+        $this->signatureImgDir = $signatureImgDir;
+    }
+
+    public function present(Signature $signature, int $languageId): array
+    {
+        $order = new Order($signature->getOrderId());
+        $customer = $order->getCustomer();
+        $gender = new Gender($customer->id_gender, $languageId);
+
+        return [
+            'firstName' => $customer->firstname,
+            'lastName' => $customer->lastname,
+            'gender' => $gender->name,
+            'filename' => $this->signatureImgDir.$signature->getFilename()
+        ];
+    }
+}
+```
+
+Let's create a twig template in `modules/demovieworderhooks/views/templates/admin/customer_signature.html.twig`:
+
+```twig
+{% extends '@Modules/demovieworderhooks/views/templates/admin/card.html.twig' %}
+
+{% block card_title %}
+  {{ 'Signature'|trans }}
+{% endblock %}
+
+{% block card_body %}
+  <div class="col-lg">
+    <div class="display-4 text-center">
+      {{ signature.gender }} {{ signature.firstName }} {{ signature.lastName }}
+    </div>
+    <div class="text-center">
+      <img src="{{ signature.filename }}" alt="">
+    </div>
+  </div>
+{% endblock %}
+```
+
+Lets add several methods to `DemoViewOrderHooks` class:
+
+```php
+    /**
+     * Get path to this module's template directory
+     */
+    private function getModuleTemplatePath(): string
+    {
+        return "@Modules/$this->name/views/templates/admin/";
+    }
+```
+```php
+    /**
+     * Displays customer's signature.
+     */
+    public function hookDisplayBackOfficeOrderActions(array $params)
+    {
+        /** @var SignatureRepository $signatureRepository */
+        $signatureRepository = $this->get('prestashop.module.demovieworderhooks.repository.signature_repository');
+
+        /** @var SignaturePresenter $signaturePresenter */
+        $signaturePresenter = $this->get('prestashop.module.demovieworderhooks.presenter.signature_presenter');
+
+        $signature = $signatureRepository->findOneBy(['orderId' => $params['id_order']]);
+
+        if (!$signature) {
+            return '';
+        }
+
+        return $this->render($this->getModuleTemplatePath() . 'customer_signature.html.twig', [
+            'signature' => $signaturePresenter->present($signature, (int) $this->context->language->id),
+        ]);
+    }
+```
+
+## Result
+
+After completing the steps above the result should be:
+
+ - Signature card:
+
+ {{< figure src="../img/view-order-hooks-demo/view-order-demo-result.png" title="Signature card" >}}
+
+Feel free to experiment with the other installed hooks!
```

Comment:
'Feel free to experiment with other hooks!' sounds better. But not sure if its needed at all, you are going to explain the other hooks in this tutorial, aren't you?

Reply:
Looks relevant to me 👍 module developers have plenty of other resources to learn about hooks. Most hooks are very standard ones, the only hooks with a specific behavior are:
- the top buttons (which expects `ActionButton` collection)
- the tab and content hook which expects specific markup to be displayed correctly

=> this can be done in another PR